app/components/filter-plaques/component.js

```diff
@@ -26,18 +26,20 @@ export default Ember.Component.extend({
       // Fetch display names
       if (ids.length > 0) {
         this.fetchAgentDetails(ids).then((data) => {
-            var displayFilters = filters.map((filter) => {
-              var value = filter.value;
-              data.forEach((agentData) => {
-                if (value === agentData.id) {
-                  value = agentData.name;
-                }
-              });
-              return {key: filter.key, value: value};
-            });
-            this.set('filters', displayFilters);
+          var displayFilters = filters.map((filter) => {
+            var value = filter.value;
+            for (let agentData of data) {
+              if (value === agentData.id) {
```

Comment:
Theres still a thing here. Say id 1 is Mary and 2 is Joe, and (wierdly) 3 is 2. if the tags are id 3, 2, then 1, we'll end up with Mary, 2, and Joe, as the names for those filters